projects/bms_carrier/test/test_cell_sense.c

```diff
@@ -0,0 +1,265 @@
+    cb = afe->settings.aux_result_cb;
+    result_arr = s_raw_readings.temps;
+  } else if (e->id == BMS_AFE_EVENT_TRIGGER_CELL_CONV) {
+    cb = afe->settings.cell_result_cb;
+    result_arr = s_raw_readings.voltages;
+  } else {
+    return false;
+  }
+
+  TEST_ASSERT_NOT_NULL(cb);
+  if (s_afe_should_fault)
+    return status_ok(event_raise_priority(EVENT_PRIORITY_HIGHEST, afe_events->fault_event, 0));
+  cb(result_arr, afe->settings.num_cells, afe->settings.result_context);
+  return status_ok(event_raise_priority(EVENT_PRIORITY_HIGHEST, afe_events->callback_run_event, 0));
+}
+
+static LtcAfeStorage s_afe;
+static StatusCode prv_init_ltc(void) {
+  status_ok_or_return(gpio_init());
+  interrupt_init();
+  soft_timer_init();
+  event_queue_init();
+  // TODO(SOFT-9): Should update this to match hardware and add missing initializations
+  LtcAfeSettings afe_settings = {
+    .num_devices = NUM_DEVICES_TO_TEST,
+    .num_cells = NUM_CELLS_TO_TEST,
+    // only need to initialize events so the correct ones are called
+    .ltc_events = { .trigger_cell_conv_event = BMS_AFE_EVENT_TRIGGER_CELL_CONV,
+                    .cell_conv_complete_event = BMS_AFE_EVENT_CELL_CONV_COMPLETE,
+                    .trigger_aux_conv_event = BMS_AFE_EVENT_TRIGGER_AUX_CONV,
+                    .aux_conv_complete_event = BMS_AFE_EVENT_AUX_CONV_COMPLETE,
+                    .callback_run_event = BMS_AFE_EVENT_CALLBACK_RUN,
+                    .fault_event = BMS_AFE_EVENT_FAULT },
+    .cell_result_cb = NULL,
+    .aux_result_cb = NULL,
+    .result_context = NULL,
+  };
+  return ltc_afe_init(&s_afe, &afe_settings);
+}
+
+void prv_check_cell_results(bool is_clear) {
+  for (size_t i = 0; i < NUM_CELLS_TO_TEST; i++) {
+    if (is_clear) {
+      TEST_ASSERT_EQUAL(s_readings.temps[i], 0);
+      continue;
+    }
+
+    TEST_ASSERT_TRUE(s_readings.voltages[i] <= VOLTAGE_READING_MAX);
+    TEST_ASSERT_TRUE(s_readings.voltages[i] >= VOLTAGE_READING_MIN);
+    s_readings.voltages[i] = 0;
+  }
+}
+
+void prv_check_temp_results(bool is_clear) {
+  for (size_t i = 0; i < NUM_CELLS_TO_TEST; i++) {
+    if (is_clear) {
+      TEST_ASSERT_EQUAL(s_readings.temps[i], 0);
+      continue;
+    }
+
+    TEST_ASSERT_TRUE(s_readings.temps[i] <= TEMP_READING_MAX);
+    TEST_ASSERT_TRUE(s_readings.temps[i] >= TEMP_READING_MIN);
+    s_readings.temps[i] = 0;
+  }
+}
+
+void setup_test(void) {
+  s_is_charging = false;
+  s_afe_should_fault = false;
+  s_expected_fault_bitset = EE_BPS_STATE_OK;
+  TEST_ASSERT_OK(prv_init_ltc());
+}
+
+void teardown_test(void) {}
+
+void prv_test_fsm_fault() {
+  Event e = { 0 };
+  LOG_DEBUG("Verifying cell result empty\n");
+  MS_TEST_HELPER_ASSERT_NEXT_EVENT_ID(e, BMS_AFE_EVENT_TRIGGER_CELL_CONV);
+  ltc_afe_process_event(&s_afe, &e);
+  prv_check_cell_results(true);
+
+  LOG_DEBUG("Verifying FSM result fault\n");
+  MS_TEST_HELPER_ASSERT_NEXT_EVENT_ID(e, BMS_AFE_EVENT_FAULT);
+  cell_sense_process_event(&e);
+}
+
+void prv_test_single_loop() {
```

Comment:
is there a reason these `prv` functions are below setup/teardown? we don't really have solid convention

Reply:
prv functions above are mocking and checks. The ones below are exported test functionality. I'll move it up since I think that's a bit unexpected but these tests are supposed to act as a type of documentation of how to use the module/what it can do. So I figured I wanted to keep "what the module is doing" close to the actual tests